src/app/home/table-data/table-data-datasource.ts

```diff
@@ -0,0 +1,111 @@
+import { DataSource } from '@angular/cdk/collections';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
+import { map } from 'rxjs/operators';
+import { Observable, of as observableOf, merge } from 'rxjs';
+
+// TODO: Replace this with your own data model type
+export interface TableDataItem {
```

Comment:
for all exported entities please add comments. 

/**
 * Interface to work with blah blah blah
 */
export interface TableDataItem {